operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/DependentResource.java

```diff
@@ -0,0 +1,62 @@
+package io.javaoperatorsdk.operator.api.reconciler.dependent;
+
+import java.util.Optional;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.config.Utils;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+
+public interface DependentResource<R, P extends HasMetadata> {
+  default EventSource initEventSource(EventSourceContext<P> context) {
+    throw new IllegalStateException("Must be implemented if not automatically provided by the SDK");
+  }
+
+  @SuppressWarnings("unchecked")
+  default Class<R> resourceType() {
+    return (Class<R>) Utils.getFirstTypeArgumentFromInterface(getClass());
+  }
+
+  default void delete(R fetched, P primary, Context context) {}
+
+  /**
+   * Computes the desired state of the dependent based on the state provided by the specified
+   * primary resource.
+   * 
+   * The default implementation returns {@code empty} which corresponds to the case where the
+   * associated dependent should never be created by the associated reconciler or that the global
+   * state of the cluster doesn't allow for the resource to be created at this point.
+   * 
+   * @param primary the primary resource associated with the reconciliation process
+   * @param context the {@link Context} associated with the reconciliation process
+   * @return an instance of the dependent resource matching the desired state specified by the
+   *         primary resource or {@code empty} if the dependent shouldn't be created at this point
+   *         (or ever)
+   */
+  default Optional<R> desired(P primary, Context context) {
+    return Optional.empty();
```

Comment:
I think it should not have a default implementation. 

The most common case is like in the WebPage, when all the resources are managed by the operator, IMHO that should be the default approach. If the resource is not managed by us (so not created and not updated) should be probably a feature flag/mode on the resources. Maybe eventually a different type of resource.

Reply:
I'm not sure I follow. If we're not creating, nor updating the resource, why would it even be a dependent?

Reply:
To get input from it, like not sure from nodes or other configurations of the cluster. But something like this happens also in the tomcat example too if remember correctly.